server/tasks/process_file.py

```diff
@@ -45,15 +47,12 @@ def process_file_streaming(self, serialized_agent: bytes):
 
 
 async def async_process_streaming_output(
-    input_job_id: str, result_handler: str, batch_size: int
+    input_job_id: str, result_handler: ResultHandler, batch_size: int
 ):
     logger.info(f"Polling for results {input_job_id=}")
 
-    try:
-        result_handler = ResultHandler.__dict__[result_handler]
-    except KeyError as e:
-        logger.error(f"{result_handler} is not a valid ResultHandler")
-        raise e
+    # FIXME switch to model_run_id or find a way to pass this in
+    result_handler.set_job_id(input_job_id)
```

Comment:
this is the awkward logic that motivates updating the corresponding LSE endpoint @hakan458